src/widget/wrecordingduration.cpp

```diff
@@ -13,10 +12,19 @@ WRecordingDuration::~WRecordingDuration() {
 void WRecordingDuration::setup(const QDomNode& node, const SkinContext& context) {
     WLabel::setup(node, context);
     connect(m_pRecordingManager, SIGNAL(durationRecorded(QString)),
-        this, SLOT(refreshLabelText(QString)));
+        this, SLOT(refreshLabel(QString)));
+    connect(m_pRecordingManager, SIGNAL(isRecordingActive(bool)),
+        this, SLOT(clearLabel(bool)));
 }
 
-void WRecordingDuration::refreshLabelText(QString durationRecorded) {
-    m_durationRecordedStr = durationRecorded;
-    setText(m_durationRecordedStr);
+void WRecordingDuration::clearLabel(bool toggle) {
```

Comment:
Th function name is misleading. It is more like "clearLebleIfInactive()" or "slotRecordingActivChanged()"